File: Array/1375. Bulb Switcher III.py
class Solution:
    def numTimesAllBlue(self, light: List[int]) -> int:
        m, b = 0, 0
        for c, i in enumerate(light, 1):
            if i > b:
                b = i
            if b == c:
                m += 1        
        return m

class Solution:
    def numTimesAllBlue(self, A):
        right = res = 0
        for i, a in enumerate(A, 1):
            right = max(right, a)
            res += right == i
        return res

File: Array/1375. Bulb Switcher III.cpp
class Solution {
public:
    int numTimesAllBlue(vector<int>& A) {
        int right = 0, res = 0, n = A.size();
        for (int i = 0; i < n; ++i)
            res += (right = max(right, A[i])) == i + 1;
        return res;
    }
};


//Not smart
class Solution {
public:
    int numTimesAllBlue(vector<int>& light) 
    {
        vector<bool> on(light.size(),false);
        int maxBlue=0,result=0;
        for(int i=0;i<light.size();i++)
        {
            on[light[i]-1]=true;
            while(maxBlue<light.size()&&on[maxBlue])   //Increase the end of blue bulbs sequence starting from 0,
                maxBlue++;
            result+=maxBlue==i+1;                  		//If number of blue bulbs is equal to number of switched on bulbs.
        }
        return result;
    }
};
